### PR TITLE
Use custom twig extensions to use manifest.json

### PR DESCRIPTION
**Pull request by Nyholm**
Blocked by https://github.com/CouscousPHP/Couscous/pull/248

UPDATE:
This PR uses a forked couscous.phar.

### CODE REVIEW

File: website/Twig/AsyncAwsTwigExtension.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        $output = '';
+
+        $entrypoints = $this->getEntrypoints();
+        if (!isset($entrypoints['entrypoints'][$name])) {
+            throw new \RuntimeException(sprintf('Could not find entrypoints.json entry for "%s"', $name));
+        }
+
+        foreach ($entrypoints['entrypoints'][$name][$prefix] as $file) {
+            $output.= sprintf($template, $file);
```
Comment (stof) [inline website/Twig/AsyncAwsTwigExtension.php]: this is missing the escaping of `$file` in case it contains special chars.

Comment (Nyholm) [reply website/Twig/AsyncAwsTwigExtension.php:103]: Can $file contain special chars? I believe `entrypoints.json` contain the correct urls

Comment (Nyholm) [reply website/Twig/AsyncAwsTwigExtension.php:103]: Maybe you are correct. But as long as we are using "app" as our entry name, arn't we safe? 

```
{
  "entrypoints": {
    "app": {
      "css": [
        "/assets/0.3639a826.css",
        "/assets/app.9f119eed.css"
      ],
      "js": [
        "/assets/0.24ed7736.js",
        "/assets/app.c7303778.js"
      ]
    }
  }
}
```

Comment (stof) [reply website/Twig/AsyncAwsTwigExtension.php:103]: Well, I would not guarantee that no chars in it need HTML escaping.

As we know our templates, we could do `htmlspecialchars($file)`

Comment (Nyholm) [reply website/Twig/AsyncAwsTwigExtension.php]: Cool. I've updated the PR.

File: website/Twig/AsyncAwsTwigExtension.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * @var array|null
+     */
+    private $entrypointsCache;
+
+    public function getFunctions()
+    {
+        return [
+            new \Twig_SimpleFunction('asset', [$this, 'getAssetUrl']),
+            new \Twig_SimpleFunction('renderStyle', [$this, 'renderStyle'], [
```
Comment (stof) [inline website/Twig/AsyncAwsTwigExtension.php:25]: use `\Twig\TwigFunction` to be foward-compatible with new Twig versions (Twig 3 uses only namespaced classes)

File: website/Twig/AsyncAwsTwigExtension.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            new \Twig_SimpleFunction('renderScript', [$this, 'renderScript'], [
+                'is_safe' => ['html'],
+            ]),
+        ];
+    }
+
+    public function getFilters()
+    {
+        return [
+            new \Twig_SimpleFilter('asset', [$this, 'parseAssetUrls']),
```
Comment (stof) [inline website/Twig/AsyncAwsTwigExtension.php:37]: same here for `\Twig\TwigFilter`

File: website/Twig/AsyncAwsTwigExtension.php (hunk 2 above)
Comment (Nyholm) [reply website/Twig/AsyncAwsTwigExtension.php:25]: Couscous is using Twig 1. Is there a forward bridge from Twig 1 so I can use namespaces?

Comment (Nyholm) [reply website/Twig/AsyncAwsTwigExtension.php:25]: Yes, it does. https://github.com/async-aws/aws/pull/427

Comment (stof) [reply website/Twig/AsyncAwsTwigExtension.php:25]: thus, you already use the namespace for the AbstractExtension base class :smile:

**Pull request merged**